server/controllers/v1/user.ts

```diff
@@ -0,0 +1,92 @@
+import argon2 from "argon2";
+import jwt from "jsonwebtoken";
+import { Request, Response } from "express";
+
+import db from "../../models/index";
+import { User } from "../../models/user";
+
+export default class UserController {
+  static secret = process.env.SECRET as string;
+  static UserModel: typeof User = db.User;
+
+  static updateLastLogin = async (user: User) => {
+    user.lastLogin = new Date();
+    await user.save();
+  };
+
+  static signIn = async (req: Request, res: Response) => {
+    const { userName, password } = req.body;
+
+    try {
+      const oldUser = await UserController.UserModel.findOne({
+        where: {
+          userName: userName,
+        },
+      });
+      if (!oldUser)
```

Comment:
Does this work if an error occurs in the `try` case? Might need a `catch` statement